### ISSUE TITLE
SA1515 (Single-line comment must be preceded by blank line) should not be reported for single line comment directly following a C# 7 case

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#2462]
When using C# 7 `case` statements (with type testing included), `SA1515` should not flag single-line comments.

The following snippet produces such a warning in StyleCop.Analyzers 1.0.2:

```csharp
static void Main(object foo)
{
    switch (foo)
    {
        case string s:
            // hi there
            return;
        default:
            break;
    }
}
```

**Comment by AArnott** [comment]
This was previously fixed in #878 but the C# 7 syntax seems to throw it off again.
